crawler.py

```diff
@@ -66,18 +66,21 @@ def generate_url(args):
     # start crawling
 
     to_crawl = deque([URL])
-    crawl_limit = math.inf if len(args) == 1 else args[1]
-
+    CRAWL_LIMIT = int(math.inf if len(args) == 1 else args[1])
     parser = MarkdownParser()
     scraper = Scraper()
 
     CRAWLS = 0
-    while to_crawl and CRAWLS < crawl_limit:
+    while to_crawl and CRAWLS < CRAWL_LIMIT:
         url = to_crawl.popleft()
         md_contents = scraper.scrape(url, user, password)
+        all_repos = set()
         for key, val in md_contents.items():
             print(f"Text of markdown file at {key} url")
-            print(parser.parse(val)[:100])
+            TEXT = parser.parse(val)
+            repos = parser.find_repos(val)
+            cleaned_repos = parser.clean(repos)
```

Comment:
Just call the clean method at the end of the find_repos method in parser, then you only need to 1 method call here.
After doing this, you should also rename the clean method in parser from `clean()` to `_clean()`, to indicate that it's a private method only meant for internal used by the parser class. See [here](https://stackoverflow.com/questions/8689964/why-do-some-functions-have-underscores-before-and-after-the-function-name).

Reply:
@deniselezi Thanks, it's done, check [0a6545f](https://github.com/deniselezi/pcs-crawler/pull/7/commits/0a6545f6c7f5133d007aeee58f3c1c7e5d781542).